src/components/settings/generalsettings/LanguageSettings.js

```diff
@@ -0,0 +1,236 @@
+  * You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+ /* eslint-disable no-undef */
+ import { Lightning, Utils, Log, Language, Storage} from '@lightningjs/sdk'
+ import { Colors } from '../../../constants/ColorConstants'
+ import { ImageConstants } from '../../../constants/ImageConstants'
+ import { GeneralSettingsTile } from './GeneralSettingsTile'
+
+
+ 
+ /**
+  * @export
+  * @class Resolution
+  * @extends Lightning.Component
+  * Renders the Resolution Component
+  */
+ export class Language_Class extends Lightning.Component {
+   /**
+    * @static
+    * @returns
+    * @memberof general settings
+    * Renders the template
+    */
+   static _template() {
+     return {
+       LanguageBg: {
+         RectangleWithGradientLeftRight: {
+           w: 960,
+           h: 1080,
+           rect: true,
+           colorLeft: Colors.DIM_BLACK,
+           colorRight: Colors.DARK_BLACK
+         },
+         BackArrow: { x: 81, y: 54, src: Utils.asset(ImageConstants.BACK_ARROW) },
+         SettingsLabel: {
+           x: 133,
+           y: 54,
+           text: {
+             text: Language.translate('Settings'),
+             fontSize: 28,
+             textColor: Colors.TRANSPARENT_GREY,
+             fontFace: 'Regular'
+           }
+         },
+         GeneralSettingsLabel: {
+           x: 82,
+           y: 113,
+           text: {
+             text: Language.translate('General_Settings'),
+             fontSize: 36,
+             textColor: Colors.LIGHTER_WHITE,
+             fontFace: 'Medium'
+           }
+         },
+         Divider: { rect: true, w: 959.5, h: 2, x: 0.5, y: 178, color: Colors.LIGHTER_BLACK },
+         DiscoverLanguage: {
+           x: 81,
+           y: 209,
+           label: Language.translate('Language Settings'),
+           type: GeneralSettingsTile
+         },
+         LanguageControl: {
+          x: 85,
+          y: 325,
+          text: {
+            text: Language.translate('Available Languages'),
+            fontSize: 34,
+            textColor: Colors.TRANSPARENT_GREY,
+            fontFace: 'Regular'
+          }
+         },
+         DiscoveredLanguage: {
+           type: Lightning.components.ListComponent,
+           x: 81,
+           y: 398,
+           w: 800,
+           h: 1080,
+           visible: false,
+           itemSize: 120,
+           horizontal: false,
+           roll: true,
+           rollMax: 1080,
+           rollMin: 0,
+           invertDirection: true
+         }
+       }
+     }
+   }
+ 
+   _init() {
+    
+     console.log("Ltag: LanguageSettings init");

```

Comment:
Discuss with Krishna Priya whether 'Log' should be used instead of console everywhere